pymedphys/_gui/streamlit/pseudonymise.py

```diff
@@ -0,0 +1,297 @@
+        zipfile_path.  Highly desirable because the zip written to zipfile_path can't
+        be deleted from this module.
+
+    """
+
+    bad_data = False
+    file_count = 0
+    keywords = pseudonymisation_api.get_default_pseudonymisation_keywords()
+    keywords.remove("PatientSex")
+    strategy = pseudonymisation_api.pseudonymisation_dispatch
+    zip_stream = zipfile_path
+    if zip_bytes_io is not None:
+        zip_stream = zip_bytes_io
+
+    with ZipFile(zip_stream, mode="w", compression=ZIP_DEFLATED) as myzip:
+        for uploaded_file_buffer in file_buffer_list:
+            file_count += 1
+
+            # don't close the buffers.  Streamlit provides the user with control over that.
+            # might be appropriate to close the buffers in some circumstances,
+            # but then when the user goes to close the buffer (click x on screen)
+            # there will be an error.
+
+            try:
+                original_file_name = uploaded_file_buffer.name
+                ds_input: pydicom.FileDataset = pydicom.dcmread(
+                    uploaded_file_buffer, force=True
+                )
+
+                anonymise_dataset(
+                    ds_input,
+                    delete_private_tags=True,
+                    delete_unknown_tags=True,
+                    copy_dataset=False,  # do the work in place.  less memory used and we're disposing shortly anyway
+                    identifying_keywords=keywords,
+                    replacement_strategy=strategy,
+                )
+                temp_anon_filepath = build_pseudonymised_file_name(ds_input)
+                anon_filename = os.path.basename(temp_anon_filepath)
+                pydicom.dcmwrite(temp_anon_filepath, ds_input)
+            except (KeyError, IOError, ValueError) as e_info:
+                print(e_info)
+                print(f"While processing {original_file_name}")
+                bad_data = True
+                break
+            myzip.write(temp_anon_filepath, anon_filename)
+            remove_file(temp_anon_filepath)
+    return bad_data
+
+
+def pseudonymise_buffer_list(file_buffer_list: list):
+    """Pseudonymises the contents of the file_buffer_list (list of DICOM files)
+    and places the pseudonymised files in to a set of zips, each less than 50 MBytes.
+    Those zips are then made available for download through a set of links in the
+    streamlit.sidebar
+
+    The 50MByte limit is imposed by the href download link limit.
+
+    The compression used is DEFLATE, but the uncompressed contents are kept at just under 50 MBytes
+    If a single original uncompressed file is > 50MByte, and does not compress to under 50 MByte
+    that file will fail to be made available for download, and may cause the entire pseudonymisation
+    attempt to fail.
+
+    Parameters
+    ----------
+    file_buffer_list : ``list``
+        DICOM files that were uploaded using streamlit.file_uploader
+
+    Returns
+    -------
+    ``list``
```

Comment:
```suggestion
    zip_path_list : list
```

See returns section of https://numpydoc.readthedocs.io/en/latest/format.html